core/domain/auth_jobs_one_off.py

```diff
@@ -236,53 +236,66 @@ def report_assoc_inconsistencies(
             str. Debug information about a discovered inconsistency.
         """
-        if assoc_by_auth_id is None and assoc_by_user_id is None:
+        if assoc_info_by_auth_id is None and assoc_info_by_user_id is None:
             # User is deleted and will be managed by the reduce() logic.
             return
 
-        distinct_user_ids = {assoc_by_auth_id[0]}
-        distinct_deleted_bools = {assoc_by_auth_id[1]}
+        user_id_of_assoc_by_auth_id, deleted_bool_of_assoc_by_auth_id = (
+            assoc_info_by_auth_id)
 
-        if assoc_by_user_id is None:
+        if assoc_info_by_user_id is None:
             yield (
                 'UserIdByFirebaseAuthIdModel(id="%s") does not correspond to a '
                 'unique UserAuthDetailsModel' % auth_id)
+            user_id_of_assoc_by_user_id = deleted_bool_of_assoc_by_user_id = (
+                None)
         else:
-            distinct_user_ids.add(assoc_by_user_id[0])
-            distinct_deleted_bools.add(assoc_by_user_id[1])
+            user_id_of_assoc_by_user_id, deleted_bool_of_assoc_by_user_id = (
+                assoc_info_by_user_id)
 
-        if len(distinct_user_ids) == 2:
+        if (user_id_of_assoc_by_user_id is not None and
+                user_id_of_assoc_by_user_id != user_id_of_assoc_by_auth_id):
             yield (
                 'auth_id="%s" has inconsistent `user_id` assignments: '
                 'UserIdByFirebaseAuthIdModel(user_id="%s") does not match '
                 'UserAuthDetailsModel(id="%s")' % (
-                    auth_id, assoc_by_auth_id[0], assoc_by_user_id[0]))
+                    auth_id,
+                    user_id_of_assoc_by_auth_id,
+                    user_id_of_assoc_by_user_id))
 
-        if len(distinct_deleted_bools) == 2:
+        if (deleted_bool_of_assoc_by_user_id is not None and
+                deleted_bool_of_assoc_by_user_id !=
+                deleted_bool_of_assoc_by_auth_id):
             yield (
                 'auth_id="%s" has inconsistent `deleted` assignments: '
                 'UserIdByFirebaseAuthIdModel(user_id="%s", deleted=%r) does '
                 'not match UserAuthDetailsModel(id="%s", deleted=%r)' % (
-                    auth_id, assoc_by_auth_id[0], assoc_by_auth_id[1],
-                    assoc_by_user_id[0], assoc_by_user_id[1]))
-
-        if firebase_account is not None:
-            _, firebase_account_is_disabled = firebase_account
-            if firebase_account_is_disabled and not any(distinct_deleted_bools):
+                    auth_id,
+                    user_id_of_assoc_by_auth_id,
+                    deleted_bool_of_assoc_by_auth_id,
+                    user_id_of_assoc_by_user_id,
+                    deleted_bool_of_assoc_by_user_id))
+
+        if firebase_account_info is not None:
+            _, firebase_account_is_disabled = firebase_account_info
+            if (firebase_account_is_disabled
+                    and not deleted_bool_of_assoc_by_auth_id
```

Comment:
What if one of these is True and the other is None?

Reply:
It's still fine. The error is only important when one of them is True, so None and False can be ignored.

Reply:
Right. So, in the case I mentioned, the error is important (per what you said), but the error won't trigger here because one of the "and" clauses will evaluate to False, right?

Reply:
Got it. I've clarified why that's not the case, PTAL.